Index: /branches/1.10/gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitedatasource.cpp
===================================================================
--- /branches/1.10/gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitedatasource.cpp	(revision 26252)
+++ /branches/1.10/gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitedatasource.cpp	(revision 26533)
@@ -553,5 +553,14 @@
             osCommand =  "SELECT InitSpatialMetadata('NONE')";
         else
-            osCommand =  "SELECT InitSpatialMetadata()";
+        {
+            /* Since spatialite 4.1, InitSpatialMetadata() is no longer run */
+            /* into a transaction, which makes population of spatial_ref_sys */
+            /* from EPSG awfully slow. We have to use InitSpatialMetadata(1) */
+            /* to run within a transaction */
+            if( OGRSQLiteGetSpatialiteVersionNumber() >= 41 )
+                osCommand =  "SELECT InitSpatialMetadata(1)";
+            else
+                osCommand =  "SELECT InitSpatialMetadata()";
+        }
         rc = sqlite3_exec( hDB, osCommand, NULL, NULL, &pszErrMsg );
         if( rc != SQLITE_OK )
